From ed1e7f5fc5c083c17bd44e4220a35eae140f980b Mon Sep 17 00:00:00 2001
From: Kai Uwe Broulik <kde@privat.broulik.de>
Date: Wed, 24 Jan 2018 09:22:53 +0100
Subject: [PATCH 3/3] [UDisks] Only ignore non-user backing file if it is known

Only loop devices have this property, otherwise we end up ignoring regular disks as well.

BUG: 389358
---
 src/solid/devices/backends/udisks2/udisksstoragevolume.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/solid/devices/backends/udisks2/udisksstoragevolume.cpp b/src/solid/devices/backends/udisks2/udisksstoragevolume.cpp
index d74eaad..623eb5d 100644
--- a/src/solid/devices/backends/udisks2/udisksstoragevolume.cpp
+++ b/src/solid/devices/backends/udisks2/udisksstoragevolume.cpp
@@ -104,5 +104,5 @@ bool StorageVolume::isIgnored() const
     }
 
     const QString backingFile = m_device->prop("BackingFile").toString();
-    return backingFile.isEmpty() || !backingFile.startsWith(QDir::homePath());
+    return !backingFile.isEmpty() && !backingFile.startsWith(QDir::homePath());
 }
-- 
2.7.4

